### ISSUE TITLE
Duplication of messages with redundant leaf-node connections

### ISSUE THREAD

**Issue by quickflash10** [nats-io/nats-server#1585]
## Defect

Make sure that these boxes are checked before submitting your issue -- thank you!

 - [ ] Included `nats-server -DV` output
 - [ ] Included a [Minimal, Complete, and Verifiable example] (https://stackoverflow.com/help/mcve)

#### Versions of `nats-server` and affected client libraries used:  

- 2.2.0-beta.12 - multiple clients (go and C#)

#### Steps or code to reproduce the issue:
I have 2 clusters, a datacenter cluster and a remote cluster. The remote cluster has 3 servers (A,B and C), and 2 of those servers (A and B) are connected to the datacenter through leaf node connection with 2 accounts. These dual leaf node connections are in place for redundancy in case one of the remote cluster servers is rebooted. What we are noticing is that if I have a client connected to Server A and publishing a message and a client connected to Server B and listening, I will get a duplication of messages. This duplication goes away if I disable one of the leaf node connections (either A or B), or publish and subscribe is happening on the same server (A to A, B to B).

Specific configuration for leafnodes on A and B

```hcl
leafnodes {
  remotes = [
    {
      urls: ["nats-leaf://XXX.XXX.XXX.XXX", "nats-leaf://YYY.YYY.YYY.YYY"]
      credentials: AccountM.creds
      account: ABXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
    },
    {
      url: ["nats-leaf://XXX.XXX.XXX.XXX", "nats-leaf://YYY.YYY.YYY.YYY"]
      credentials: AccountN.creds
      account: ACYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYY                         
    },
  ]
}
```

#### Expected result:

Publish one message on A, Receive one Message on B

#### Actual result:

Publish one message on A, Receive two Messages on B

**Comment by wallyqs** [comment]
Hi @quickflash10 could you try with a version `2.2.0-beta.15`? I think the change from this PR #1476 could help

**Comment by kozlovic** [comment]
Actually OP posts that server is 2.2.0-beta.22, but I am not able to reproduce this with current main branch.
I believe addition of cluster names was to fix this issue. So @quickflash10 do you see mention of cluster name in your A and B servers log? If one is not provided they should negotiate a common cluster name. If you don't see that it probably means that you are not running with the version you think you are.

**Comment by quickflash10** [comment]
Confirmed the version I was using did not have cluster names as mentioned above.  Updated to beta.23 and see the cluster name information.  My quick test is that it seems to have resolved the issue, but waiting for confirmation from a team-mate.

**Comment by quickflash10** [comment]
Issue is solved by #1476

**Comment by wallyqs** [comment]
That's great, thanks for the update @quickflash10
